src/main/java/tigase/xmpp/impl/SaslAuth.java

```diff
@@ -114,41 +83,15 @@ public void process(final Packet packet, final XMPPResourceConnection session, f
 			}
 			final String clientIp = BruteForceLockerBean.getClientIp(session);
 			if (session.isAuthorized()) {
-
-				// Multiple authentication attempts....
-				// Another authentication request on already authenticated
-				// connection
-				// This is not allowed and must be forbidden.
-				Packet res = packet.swapFromTo(createReply(ElementType.failure, "<not-authorized/>"), null, null);
-
-				// Make sure it gets delivered before stream close
-				res.setPriority(Priority.SYSTEM);
-				results.offer(res);
-
-				// Optionally close the connection to make sure there is no
-				// confusion about the connection state.
-				results.offer(Command.CLOSE.getPacket(packet.getTo(), packet.getFrom(), StanzaType.set,
-													  session.nextStanzaId()));
-				if (log.isLoggable(Level.FINEST)) {
-					log.log(Level.FINEST, "Discovered second authentication attempt: {0}, packet: {1}",
-							new Object[]{session.toString(), packet.toString()});
-				}
-				try {
-					session.logout();
-				} catch (NotAuthorizedException ex) {
-					log.log(Level.FINER, "Unsuccessful session logout: {0}", session.toString());
-				}
-				if (log.isLoggable(Level.FINEST)) {
-					log.log(Level.FINEST, "Session after logout: {0}", session.toString());
-				}
+				processSessionAlreadyAuthorized(packet, session, results);
 				return;
 			} else {
 				Element request = packet.getElement();
 				try {
 					SaslServer ss;
 
 					if ("auth" == request.getName()) {
-						final String mechanismName = request.getAttributeStaticStr("mechanism");
+						final String mechanismName = packet.getElement().getAttributeStaticStr("mechanism");
```

Comment:
I think this change is unnecessary as we already have element instance `request`? so why this changes?  if it should be made, then maybe we should replace `request.getName()` in a line above as well?

Reply:
I was just doing diff on `process()` method to see difference and it's a leftover.

Reply:
OK